guide/slash-commands/response-methods.md

```diff
@@ -0,0 +1,230 @@
+		.setName('ping')
+		.setDescription('Replies with Pong!'),
+	async execute(interaction) {
+		await interaction.reply('Pong!');
+	},
+};
+```
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="user" :command="true">ping</DiscordInteraction>
+		</template>
+		Pong!
+	</DiscordMessage>
+</DiscordMessages>
+
+::: warning
+Initially an interaction token is only valid for three seconds, so that's the timeframe in which you are able to use the `ChatInputCommandInteraction#reply()` method. Responses that require more time ("Deferred Responses") are explained later in this page.
+:::
+
+## Ephemeral responses
+
+You may not always want everyone who has access to the channel to see a slash command's response. Previously, you would have had to DM the user to achieve this, potentially encountering the high rate limits associated with DM messages, or simplying being unable to if the user's DMs are disabled. 
+
+Thankfully, Discord provides a way to hide resposne messages from everyone but the executor of the slash command. This type of message is called an `ephemeral` message and can be set by providing `ephemeral: true` in the `InteractionReplyOptions`, as follows:
+
+```js {5}
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isChatInputCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.reply({ content: 'Secret Pong!', ephemeral: true });
+	}
+});
+```
+
+Now when you run your command again, you should see something like this:
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction
+				profile="user"
+				:command="true"
+				:ephemeral="true"
+			>ping</DiscordInteraction>
+		</template>
+		Pong!
+	</DiscordMessage>
+</DiscordMessages>
+
+Ephemeral responses are *only* available for interaction responses; another great reason to switch across to the new and improved features.
+
+## Editing responses
+
+After you've sent an initial response, you may want to edit that response for various reasons. This can be achieved with the `ChatInputCommandInteraction#editReply()` method:
+
+::: warning
+After the initial response, an interaction token is valid for 15 minutes, so this is the timeframe in which you can edit the response and send follow-up messages. You also **cannot** edit the ephemeral state of a message, so ensure that your first response sets this correctly.
+:::
+
+```js {1,8-9}
+const wait = require('node:timers/promises').setTimeout;
+
+client.on('interactionCreate', async interaction => {
+	if (!interaction.isChatInputCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		await interaction.reply('Pong!');
+		await wait(2000);
+		await interaction.editReply('Pong again!');
+	}
+});
+```
+
+In fact, editing your interaction response is necessary to [calculate the ping](popular-topics/faq#how-do-i-check-the-bot-s-ping) properly for this command.
```

Comment:
404?

Reply:
Worked on my machine! I'll check it again